### PR TITLE
Include surface ocean current in the computation of air-sea fluxes

### LINKED ISSUE #312: Downscaling sea surface current field from parent grid to nested grid
**Is your feature request related to a problem? Please describe.**
The surface ocean current will be included in the computation of air-sea fluxes in both parent grid and nested grid. The code needs to be updated to transfer the surface current filed from parent grid to the nested grid.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
Related PR: #311

### PR DESCRIPTION
**Pull request by binli2337**
**Description**

Update surface layer scheme to include surface ocean current in the computation of air sea fluxes.
Transfer surface ocean current field to nested grids.

Fixes #312 

**How Has This Been Tested?**

The updated code has been tested successfully in the HAFS-MOM6 application.

**Checklist:**

Please check all whether they apply or not
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

Co-authored-by @binli2337 ,@BinLiu-NOAA, @weiguoWang-NOAA, @BijuThomas-NOAA, @hyunsookkim-NOAA, @XuLi-NOAA, @MariaAristizabal-NOAA, @JohnSteffen-NOAA

### CODE REVIEW

**Comment by lharris4** [comment]
Hi, @binli2337 . I take it this is to be able to transmit the surface currents between the parent and nested grids, something that the physics itself does not support? Is this capability needed for other variables as well?

Thanks,
Lucas

**Review by lharris4 — APPROVED** [review]
I think this is OK given that it extends a previously defined capability used for operations. However I really think this is something the flux coupler needs to handle, by transmitting surface ocean conditions (SST, SSH, Us, Vs, roughness length, etc.) to the appropriate domain, nested or not. 

I would appreciate if you would open an issue within the coupling library.

Thanks,
Lucas

File: model/fv_arrays.F90 [hunk 1]
```diff
@@ -1269,6 +1269,8 @@ module fv_arrays_mod
     real, _ALLOCATABLE :: sgh(:,:)      _NULL  !< Terrain standard deviation
     real, _ALLOCATABLE :: oro(:,:)      _NULL  !< land fraction (1: all land; 0: all water)
     real, _ALLOCATABLE :: ts(:,:)       _NULL  !< skin temperature (sst) from NCEP/GFS (K) -- tile
+    real, _ALLOCATABLE :: usfco(:,:)    _NULL  !< sea surface current (zonal) from NCEP/GFS (m s-1) -- tile
+    real, _ALLOCATABLE :: vsfco(:,:)    _NULL  !< sea surface current (meridional) from NCEP/GFS (m s-1) -- tile
```
Comment (bensonr) [inline model/fv_arrays.F90:1273]: Are these necessary?  I don't see them being used in your feature branch.

Comment (binli2337) [reply model/fv_arrays.F90:1273]: @bensonr The variables usfco and vsfco are used in the updated surface layer scheme in ccpp-physics. The updated surface layer scheme is used in both parent and nested grids,

Comment (BinLiu-NOAA) [reply model/fv_arrays.F90:1273]: @bensonr I think this is a good catch. The two lines of defining usfco/vsfco are not needed here inside the FV3dycore. The two variables are needed in the FV3ATM level though (including inside the moving_nest dir). @binli2337 is currently double checking and testing, will reply to this thread once he confirms. Thanks!

**Review by bensonr — APPROVED** [review]

**Review by BinLiu-NOAA — APPROVED** [review]
Changes look good to me.

**Comment by lharris4** [comment]
Hi, Bin. Thanks for opening this issue.

Lucas

**Comment by zach1221** [comment]
Testing is complete on ufs-wm 2104. @bensonr would you be able to merge this PR for us please?

**Pull request merged**